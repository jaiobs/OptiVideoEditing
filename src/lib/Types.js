export const ORIENTATION = 'orientation';
export const PICTURE_TAKEN = 'picture_taken';
export const VIDEO_TAKEN = 'video_taken';
export const BACK_PRESS_VIDEO = 'back_press_video';

export const STICKER_CATEGORIES_SUCCESS = 'Sticker_categories_Success';
export const STICKER_CATEGORIES_FAILURE = 'Sticker_categories_Failure';
export const STICKER_LIST_SUCCESS = 'Sticker_list_Success';
export const STICKER_LIST_FAILURE = 'Sticker_list_Failure';
export const FLASH_ACTION = 'flash_action';
export const SWITCH_CAMERA_ACTION = 'switch_camera_action';

// STICKER URL

//----Fetch all sticker categories-----//
export const STICKER_CATOGORIES = 'v0/post/sticker-category-listing';

//----Fetch all stickers list based on categories------//
export const FETCH_ALL_STICKERS = 'v0/post/sticker-listing?id=';

//MUSIC SEARCH
export const TRACK_LIST_FETCH = 'track_list_fetch';

export const SOUND_CLOUD_BASE_URL = 'https://api.soundcloud.com';
export const GET_TRACKS = '/tracks';

// TAGGING
export const TAG_ALL_USERS_LISTING = 'v0/user/listing-n-search';
export const TAG_FRIENDS_LISTING = 'v0/user/friends-listing';
export const TAG_ALL_USERS_FETCH = 'v0/tag_all_users_fetch';

//RETAIN VIDEO IN PREVIEW PAGE
export const RETAIN_VIDEO_SEGMENTS = 'retain_video_segments';

//OVERLAY DATA
export const OVERLAY_DATA_LIST = 'overlay_data_list';
