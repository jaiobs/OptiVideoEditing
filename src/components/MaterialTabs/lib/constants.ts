export default {
  barHeight: 48,
  indicatorHeight: 4,
};
