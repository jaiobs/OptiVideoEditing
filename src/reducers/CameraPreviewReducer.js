import {
  ORIENTATION,
  PICTURE_TAKEN,
  VIDEO_TAKEN,
  RETAIN_VIDEO_SEGMENTS,
  BACK_PRESS_VIDEO,
  FLASH_ACTION,
  SWITCH_CAMERA_ACTION,
} from '../lib/Types';

const INITIAL_STATE = {
  orientationCheck: 'portrait',
  onPictureTaken: {},
  onVideoTaken: {},
  retainVideoSegments: [],
  backPressVideoDetails: {},
  flashDetails: false,
  switchCameraDetails: false,
};

export default (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case ORIENTATION:
      return {
        ...state,
        orientationCheck: action.payload,
      };

    case PICTURE_TAKEN:
      return {
        ...state,
        onPictureTaken: action.payload,
      };

    case VIDEO_TAKEN:
      return {
        ...state,
        onVideoTaken: action.payload,
      };

    case BACK_PRESS_VIDEO:
      return {
        ...state,
        backPressVideoDetails: action.payload,
      };

    case RETAIN_VIDEO_SEGMENTS:
      return {
        ...state,
        retainVideoSegments: action.payload,
      };

    case FLASH_ACTION:
      return {
        ...state,
        flashDetails: action.payload,
      };

    case SWITCH_CAMERA_ACTION:
      return {
        ...state,
        switchCameraDetails: action.payload,
      };

    default:
      return state;
  }
};
