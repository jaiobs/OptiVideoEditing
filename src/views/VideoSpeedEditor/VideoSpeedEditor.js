import React, {Component} from 'react';
import {
  View,
  StyleSheet,
  BackHandler,
  TouchableOpacity,
  Image,
} from 'react-native';
import {AdjustableExoPlayer, VideoTimeLineView} from '../../libs/litpic_sdk';
import BouncyView from '../../components/widgets/BouncyView/BouncyView';
import {connect} from 'react-redux';
import {orientation} from '../../actions/cameraPreviewAction';
import {Loader} from '../../components/ViewComponents/Loader';
import {Images} from '../../res';

class VideoSpeedEditor extends Component {
  constructor(props) {
    super(props);
    this.goBack = this.goBack.bind(this);
    this.state = {
      videoPath: props.navigation.getParam('videoPath', null),
      videoDetails: props.navigation.getParam('videoDetails', null),
      editedVideoPath: null,
      isVideoProcessing: false,
      previewPlaybackParams: null,
    };
  }

  componentDidMount() {
    try {
      BackHandler.addEventListener('hardwareBackPress', this.goBack);
    } catch (e) {
      console.log('catch->', e);
    }
  }

  componentWillUnmount() {
    this.releaseNativeListeners();
    try {
      BackHandler.removeEventListener('hardwareBackPress', this.goBack);
    } catch (e) {
      console.log('catch->', e);
    }
  }

  goBack() {
    try {
      if (!this.state.isVideoProcessing) {
        this.props.navigation.navigate('VideoEditor', {
          videoPath: this.state.videoPath,
          videoDetails: this.state.videoDetails,
          canMoveNext: false,
          finalPlayer: true,
        });
      }
      return true;
    } catch (e) {
      console.log('goback->', e);
    }
  }

  saveVideo() {
    console.log('saveVideo');
    this.videoTimeLineView.saveVideo();
  }

  exportVideo(event) {
    this.props.navigation.navigate('VideoEditor', {
      videoPath: event.videoPath,
      videoDetails: {},
      canMoveNext: false,
      finalPlayer: true,
    });
  }

  onNext() {
    if (this.state.previewPlaybackParams) {
      this.videoTimeLineView.onNext(this.state.previewPlaybackParams);
    } else {
      this.props.navigation.navigate('VideoEditor', {
        videoPath: this.state.videoPath,
        videoDetails: {},
        canMoveNext: false,
        finalPlayer: true,
      });
    }
  }

  releaseNativeListeners() {
    if (this.videoPlayer != null) {
      this.videoPlayer.releaseListeners();
    }
  }

  seekTo(event) {
    console.log('seek to ', event);
    if (this.videoPlayer != null) {
      this.videoPlayer.seekTo(event);
    }
  }

  updatePreview(event) {
    console.log('update preview ', event.videoPath);
    this.setState({editedVideoPath: event.videoPath});
    if (this.videoPlayer != null) {
      this.videoPlayer.updatePreview(event);
    }
  }

  updateVideoPlayback(event) {
    if (this.videoPlayer != null) {
      this.setState({previewPlaybackParams: event});
      this.videoPlayer.updatePreview(event);
    }
  }

  showOrHideLoader(showLoader) {
    this.setState({isVideoProcessing: showLoader});
  }

  render() {
    return (
      <View style={styles.container}>
        <BouncyView
          onPress={() => {
            this.goBack();
          }}>
          <View style={styles.backIcons}>
            <Image source={Images.close_icon} style={styles.backIcon} />
          </View>
        </BouncyView>
        <View style={styles.portraitView}>
          <View style={styles.portraitVideo}>
            <AdjustableExoPlayer
              ref={(ref) => (this.videoPlayer = ref)}
              style={styles.playerStyle}
              videoPath={this.state.videoPath}
              videoDetails={this.state.videoDetails}
              cropPosition={(xPosition) =>
                this.setState({cropPosition: xPosition})
              }
              loaderState={this.state.isVideoProcessing}
            />
          </View>
          <View style={styles.portraitTimeLine}>
            <VideoTimeLineView
              ref={(ref) => (this.videoTimeLineView = ref)}
              videoPath={this.state.videoPath}
              seekTo={(event) => this.seekTo(event)}
              updatePreview={(event) => this.updatePreview(event)}
              updateVideoPlayback={(event) => this.updateVideoPlayback(event)}
              exportVideo={(event) => this.exportVideo(event)}
              //videoDuration={this.state.videoDetails.videoDuration}
              style={styles.timerLinePortrait}
              showOrHideLoader={(showLoader) =>
                this.showOrHideLoader(showLoader)
              }
            />
          </View>
        </View>

        <View style={styles.actionContainerSave}>
          <TouchableOpacity onPress={() => this.onNext()}>
            <Image style={styles.imageIcon} source={Images.next_video_icon} />
          </TouchableOpacity>
        </View>
        <Loader visibility={this.state.isVideoProcessing} />
      </View>
    );
  }
}
const mapStateToProps = (state) => {
  const {orientationCheck} = state.CameraPreviewReducer;
  return {
    orientationCheck,
  };
};
export default connect(mapStateToProps, {
  orientation,
})(VideoSpeedEditor);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#000',
  },
  backIcon: {
    paddingTop: 10,
    height: 20,
    width: 20,
  },
  playerStyle: {
    width: '100%',
    height: '100%',
  },
  timerLine: {
    width: '100%',
    height: 250,
  },
  timerLinePortrait: {
    width: '100%',
    height: 180,
  },
  portraitTimeLine: {
    marginLeft: 10,
    marginRight: 10,
    height: 200,
  },
  portraitView: {
    flex: 1,
  },
  portraitVideo: {
    flex: 4,
    margin: 20,
  },
  adjustableView: {
    justifyContent: 'center',
    flex: 1,
    margin: 10,
  },
  ViewStyle: {
    flex: 1,
    flexDirection: 'row',
  },
  timerLineView: {
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
  },
  backIcons: {
    paddingTop: 10,
    paddingLeft: 10,
  },
  actionContainerSave: {
    height: 40,
    alignSelf: 'flex-end',
  },
  imageIcon: {height: 25, width: 25, margin: 10},
});
