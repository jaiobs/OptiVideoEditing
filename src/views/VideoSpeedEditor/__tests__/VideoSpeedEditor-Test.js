import React from 'react';
import renderer from 'react-test-renderer';
import VideoSpeedEditor from '../VideoSpeedEditor';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import CameraPreviewReducer from '../../../reducers/CameraPreviewReducer';
import * as types from '../../../lib/Types';
jest.useFakeTimers();
const mockOrientation = 'portrait';
const mockStore = configureMockStore({});

describe('orientationCheck reducer', () => {
  jest.useFakeTimers();
  it('should return the initial state', () => {
    expect(CameraPreviewReducer(undefined, {})).toEqual({
      orientationCheck: mockOrientation,
      backPressVideoDetails: {},
      flashDetails: false,
      onPictureTaken: {},
      onVideoTaken: {},
      retainVideoSegments: [],
      switchCameraDetails: false,
    });
  });

  it('should handle orientation action payload', async () => {
    jest.useFakeTimers();
    expect(
      CameraPreviewReducer('', {
        type: types.ORIENTATION,
        payload: mockOrientation,
      }),
    ).toEqual({
      orientationCheck: mockOrientation,
    });
  });
});

describe('VideoTrimmerTest', () => {
  jest.useFakeTimers();
  it('should render correctly with state to props', async () => {
    const initialState = {
      CameraPreviewReducer: {orientationCheck: mockOrientation},
    };
    const store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <VideoSpeedEditor navigation={{getParam: jest.fn()}} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
