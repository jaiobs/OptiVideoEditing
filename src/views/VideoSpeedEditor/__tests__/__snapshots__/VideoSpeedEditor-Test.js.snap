// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoTrimmerTest should render correctly with state to props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000",
      "flex": 1,
    }
  }
>
  <View
    delay={10}
    isSpeedControl={false}
    moveSlop={15}
    onPress={[Function]}
    scale={1.1}
    style={
      Object {
        "marginBottom": 10,
        "opacity": 0,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": 45,
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
    >
      <View
        style={
          Object {
            "paddingLeft": 10,
            "paddingTop": 10,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/res/assets/images/cross.png",
            }
          }
          style={
            Object {
              "height": 20,
              "paddingTop": 10,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
          "margin": 20,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <AdjustableExoVideoPlayer
          cropPosition={[Function]}
          loaderState={false}
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "height": 200,
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      <View
        style={
          Object {
            "bottom": 0,
            "flex": 1,
            "left": 4,
            "position": "absolute",
            "right": 4,
            "zIndex": 10,
          }
        }
      >
        <VideoTimeLineView
          exportVideo={[Function]}
          seekTo={[Function]}
          showOrHideLoader={[Function]}
          style={
            Object {
              "height": 180,
              "width": "100%",
            }
          }
          updatePreview={[Function]}
          updateVideoPlayback={[Function]}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "flex-end",
        "height": 40,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/res/assets/images/next_button_send.png",
          }
        }
        style={
          Object {
            "height": 25,
            "margin": 10,
            "width": 25,
          }
        }
      />
    </View>
  </View>
</View>
`;
